### PR TITLE
[Merged by Bors] - feat: restriction of the non-unital continuous functional calculus to a scalar subring

### PR DESCRIPTION
**Pull request by j-loreaux**
Co-authored by: @ADedecker 

- [x] depends on: #13323 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean [hunk 1]
```diff
@@ -157,3 +156,147 @@ lemma cfc_eq_restrict (f : C(S, R)) (halg : UniformEmbedding (algebraMap R S)) {
...
 end SpectrumRestricts
+
+
+namespace QuasispectrumRestricts
+
+local notation "σₙ" => quasispectrum
+open ContinuousMapZero Set
+
+/-- The homeomorphism `spectrum S a ≃ₜ spectrum R a` induced by `SpectrumRestricts a f`. -/
+def homeomorph {R S A : Type*} [Semifield R] [Field S] [NonUnitalRing A]
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean:167]:
```suggestion
/-- The homeomorphism `quasispectrum S a ≃ₜ quasispectrum R a` induced by `QuasispectrumRestricts a f`. -/
def homeomorph {R S A : Type*} [Semifield R] [Field S] [NonUnitalRing A]
```

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean [hunk 2]
```diff
@@ -157,3 +156,147 @@ lemma cfc_eq_restrict (f : C(S, R)) (halg : UniformEmbedding (algebraMap R S)) {
...
+  right_inv x := Subtype.ext <| h.left_inv x
+  continuous_toFun := continuous_induced_rng.mpr <| f.continuous.comp continuous_induced_dom
+  continuous_invFun := continuous_induced_rng.mpr <|
+    continuous_algebraMap R S |>.comp continuous_induced_dom
+
+universe u v w
+
+open ContinuousMapZero
+/-- If the spectrum of an element restricts to a smaller scalar ring, then a non-unital continuous
+functional calculus over the larger scalar ring descends to the smaller one. -/
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean:184]:
```suggestion
/-- If the quasispectrum of an element restricts to a smaller scalar ring, then a non-unital continuous
functional calculus over the larger scalar ring descends to the smaller one. -/
```

File: Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean [hunk 3]
```diff
@@ -157,3 +156,147 @@ lemma cfc_eq_restrict (f : C(S, R)) (halg : UniformEmbedding (algebraMap R S)) {
...
+  convert h_id
+  ext x
+  exact h.rightInvOn x.2
+
+variable [IsScalarTower R A A] [SMulCommClass R A A]
+
+/-- Given a `NonUnitalContinuousFunctionalCalculus S q`. If we form the predicate `p` for `a : A`
+characterized by: `q a` and the spectrum of `a` restricts to the scalar subring `R` via
+`f : C(S, R)`, then we can get a restricted functional calculus
+`NonUnitalContinuousFunctionalCalculus R p`. -/
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean:230]:
```suggestion
/-- Given a `NonUnitalContinuousFunctionalCalculus S q`. If we form the predicate `p` for `a : A`
characterized by: `q a` and the quasispectrum of `a` restricts to the scalar subring `R` via
`f : C(S, R)`, then we can get a restricted functional calculus
`NonUnitalContinuousFunctionalCalculus R p`. -/
```

**Review by dupuisf — COMMENTED** [review]
Looks good!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**